### DISCUSSION TITLE
Can middleware restrict access to static files stored in the public folder in Next.js?

### DISCUSSION THREAD

**Discussion by mhmdsalahsebai** [vercel/next.js#72515] [Help] [answered]
### Summary

I'm building a Next.js app where I need to store static files, like videos and images, in the public folder and give them unique names using UUIDs generated with crypto. My goal is to restrict access to these files based on user permissions, so I'm considering using middleware to control access.

I understand that relying on UUIDs alone is considered 'security through obscurity'. My question is specifically about the middleware: can it reliably prevent unauthorized access to files in the public folder?

**Comment by icyJoseph** [comment, marked as answer]
Yes, somewhat.

- https://stackblitz.com/edit/nextjs-p2jcim?file=middleware.ts,public%2Fhello.txt

In that Stackblitz, the middleware checks for access to a file in the public directory, called `hello.txt`.

If there is an allow query in the url, then it shows the file, otherwise, it redirects to `/`

However, the thing to notice here, is that there's not necessarily a way to tell if the path hit by the request is in the `public` folder, at least, not that I am aware of... so you'd need to have a way to match those paths for the files you want to "hide".
